src/ae.c

```diff
@@ -63,6 +62,17 @@
 #endif
 #endif
 
+#define AE_LOCK(eventLoop)                                                                                             \
+    int __ae_locked = 0;                                                                                               \
```

Comment:
why do we need this automatic variable? isn't the AE_PROTECT_POLL enough?

Reply:
I add it as a protection against unlock without lock, now when we use `PTHREAD_MUTEX_ERRORCHECK` we can remove it, removed.